gum_test.html

```diff
@@ -287,6 +312,38 @@ <h4>getUserMedia is missing!</h4>
       setTimeout(captureImage, 2000);
     }
   }
+
+  function startEqualizer(stream) {
+    const audioCtx = new AudioContext();
+    const analyser = audioCtx.createAnalyser();
+    audioCtx.createMediaStreamSource(stream).connect(analyser);
+
+    const canvasCtx = equalizer.getContext("2d");
+    const data = new Uint8Array(equalizer.width);
+
+    let interval = setInterval(() => {
+      if (stream.getAudioTracks()[0].readyState != "live") {
+        clearInterval(interval);
```

Comment:
Or use the ended event on the track to clear the interval. Also close the AudioContext at that point please.